helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -51,10 +51,9 @@
   private Map<String, Integer> _maxCapacity;
   private int _maxPartition; // maximum number of the partitions that can be assigned to the node.
 
-  // proposed assignment tracking
-  // <resource name, partition name set>
-  private Map<String, Set<String>> _currentAssignments;
-  // <resource name, top state partition name>
+  // A map of <resource name, <partition name, state>> to track all the proposed partition assignment with state.
+  private Map<String, Map<String, String>> _currentAssignments;
+  // An index map of <resource name, top state partition name> for tracking the top state partitions.
```

Comment:
1. What do you mean by an "index map"?
2. When you say <resource name, top state partition name>, what is the top state partition name? Each resource maps to a Set of Strings, and these strings are names of nodes that have top state partitions? Could you clarify?

Reply:
Maybe just change to "index" instead of "index map"?

Some background here, this is mainly for search (that's why I take it as an index). Note we can always derive this map from the _currentAssignments. But repeatedly calculating it is not efficient.

In addition, not sure how you are confused with the node name in this context. But this set is for "the partition names of replicas that are in the top state".  Maybe I shall just put all this in the comment.